### PR TITLE
[BOUNTY] Log item cremation and deep frying

### LINKED ISSUE #1980: [Code Bounty] Add logging of items destroyed by cremator
Currently cremation is logged in attack.log

`[2020-06-15 22:25:18.313] ATTACK: bloons3/(Wesley Richardson) has cremated @Bloons3[DC]/(Ashlyn Driggers) (NEWHP: 100)  (CentCom Recovery Ship (196, 83, 1))`

However, it does not appear that item destruction is logged. So it might be possible to destroy a unique antag objective item out of spite, and have it go un-logged.

The goal of this code bounty will be to log what items are destroyed in the cremator (recursively if an inventory is destroyed) to the appropriate log file, and if possible associate the destruction of those items with a ckey.

Possible format

`[2020-06-15 22:25:18.313] ATTACK: bloons3/(Wesley Richardson) has cremated ITEM/gun/energy/laser/captain  (CentCom Recovery Ship (196, 83, 1))`

Possible alternate format:

`[2020-06-15 22:25:18.313] ATTACK: bloons3/(Wesley Richardson) has activated a crematorium  (CentCom Recovery Ship (196, 83, 1))`
`[2020-06-15 22:25:18.313] ATTACK: ITEM/gun/energy/laser/captain destroyed by crematorium  (CentCom Recovery Ship (196, 83, 1))
`

This bounty will be 15 USD payable via Cryptocurrency or BountySource, depending on which is easier for the coder who accepts.

### PR DESCRIPTION
**Pull request by ike709**
Logs item cremation. Fixes #1980 Also logs deep frying.

## Changelog
:cl:
add: Item cremation is now logged.
add: Deep frying is now logged.
/:cl:

### CODE REVIEW

**Comment by ike709** [comment]
bloons asked me to mention that they paid.

they paid.

File: code/modules/food_and_drinks/kitchen_machinery/deep_fryer.dm [hunk 1]
```diff
@@ -101,6 +101,7 @@ God bless America.
 			return ..()
 		else if(!frying && user.transferItemToLoc(I, src))
 			to_chat(user, "<span class='notice'>You put [I] into [src].</span>")
+			log_game("[key_name(user)] deep fried [I.name] ([I.type]) at [AREACOORD(src)].")
```
Comment (PowerfulBacon) [inline code/modules/food_and_drinks/kitchen_machinery/deep_fryer.dm:104]: Would be useful if this was user.log_messaged aswell, but its not required

**Review by PowerfulBacon — DISMISSED** [review]
Fills the logs up a lot, but is very, very useful and seems to work fine when tested.

**Comment by Crossedfall** [comment]
I'm almost tempted to say that deepfrying should be relegated to a dedicated cooking log or something. Just considering that this is likely going to add even more bloat to the already chunky game.log

**Comment by PowerfulBacon** [comment]
> 
> 
> I'm almost tempted to say that deepfrying should be relegated to a dedicated cooking log or something. Just considering that this is likely going to add even more bloat to the already chunky game.log

As an addendum to this, cremating creates a lot of logs, since most objects will create multiple logs for cremating.
Cremate someone?
 - body destroyed
 - glasses
 - boots
 - jumpsuit
 - suit
 - head
 - gloves
 - neckwear
 - backpack
 - ID
 - PDA
 - Everything in the backpack
 - Guns contain like 4 different components alone (firing pins etc.)

All in the logs for cremating 1 person

**Comment by ike709** [comment]
listen I just do what the admins ask for

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**